io/io/src/TBufferJSON.cxx

```diff
@@ -3232,7 +3233,7 @@ void TBufferJSON::CompactFloatString(char* sbuf, unsigned len)
 void TBufferJSON::JsonWriteBasic(Float_t value)
 {
    char buf[200];
-   if (value == floor(value)) {
+   if ((value == floor(value)) && (TMath::Abs(value) < 1e15)) {
```

Comment:
Consider
```
std::abs(std::nearbyint(value) - value) > 0.00001f)
```

Reply:
I made new commit, but main condition remains:

    value == std::nearbyint(value)

Reply:
>  but main condition remains:

Isn't it wrong?  Testing floating point for equality is numerically unstable, isn't it?